plugins/topology/README.md

```diff
@@ -1,44 +1,56 @@
 # Topology plugin for Backstage
 
-This plugin will help with visualizing the workloads like Deployment, Replicaset and Pods powering any service on the Kubernetes cluster.
+The Topology plugin helps with visualizing the workloads such as Deployment, Replicaset, and Pods powering any service on the Kubernetes cluster.
 
 ## Prerequisites
 
-1. Install and configure Kubernetes plugin by following these guides:
-   https://backstage.io/docs/features/kubernetes/installation
-   https://backstage.io/docs/features/kubernetes/configuration
-2. Kubernetes plugin is properly configured and able to connect to the cluster via a `ServiceAccount`
-3. Following [ClusterRole](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) must be granted to ServiceAccount accessing the cluster (This role is already granted if you have the Backstage Kubernetes Plugin properly configured)
-4. To get the resources from a k8s cluster add the following annotations to the entity's catalog-info.yaml as K8s plugin identifies if the provided entity has k8s resources and if yes from which namespace it should get the resources based on the following annotations.
+1. Install and configure the Kubernetes plugin by following the [installation](https://backstage.io/docs/features/kubernetes/installation) and [configuration](https://backstage.io/docs/features/kubernetes/configuration) guides.
+2. The Kubernetes plugin is configured and able to connect to the cluster using a `ServiceAccount`.
+3. The [`ClusterRole`](https://backstage.io/docs/features/kubernetes/configuration#role-based-access-control) must be granted to `ServiceAccount` accessing the cluster. If you have the Backstage Kubernetes plugin configured, then the `ClusterRole` is already granted.
+4. To receive the resources from a Kubernetes cluster, add the following annotations to the entity's `catalog-info.yaml` file:
+
    ```yaml
    annotations:
      backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
      backstage.io/kubernetes-namespace: <RESOURCE_NS>
    ```
-5. A custom label selector can also be added which will then be used by Backstage to find the resources. The label selector takes precedence over the id annotation.
+
+   The Kubernetes plugin identifies if the provided entity contains Kubernetes resources and from which namespace the plugin receives the resources based on the previous annotations.
```

Comment:
This is formulated in an ambiguous way. Does it "identify... based on the previous annotations" or  are he resources themselves, or the way the plugin receives them, "based on the previous annotations"?

Reply:
yeah this needs an update, this requirement comes from k8s plugin and details can be seen here https://backstage.io/docs/features/kubernetes/configuration/#common-backstageiokubernetes-id-label

Reply:
Can you do an initial rewrite here?

Reply:
i can give it a try and similar steps are there for tekton as well https://github.com/janus-idp/backstage-plugins/pull/265/files

so we might need to update tekton as well

Reply:
4. In order for Backstage to detect that an entity has Kubernetes resources, the following annotation should be added to the entity's `catalog-info.yaml`:
   ```yaml
   annotations:
     backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>
   ```
5. Entity's `catalog-info.yaml` can have the `backstage.io/kubernetes-namespace` annotation, this will cause the entity's Kubernetes resources to be looked up via that namespace.
   ```yaml
   annotations:
     backstage.io/kubernetes-namespace: <RESOURCE_NS>
   ```
6. A custom label selector can also be added to the entity's `catalog-info.yaml` which will then be used by Backstage to find the resources. The label selector takes precedence over the id annotation.
   `'backstage.io/kubernetes-label-selector': 'app=my-app,component=front-end`
   ```yaml
   annotations:
     ...
     backstage.io/kubernetes-label-selector: 'app=my-app,component=front-end'
   ```
7. For the Kubernetes plugin to get the desired entity's Kubernetes resources the following label should be added to the resources(if a label selector is used then the labels mentioned in that should be present on the resource):
   ```yaml
   labels:
     ...
     backstage.io/kubernetes-id: <BACKSTAGE_ENTITY_NAME>`
   ```
8. Workload resources like Deployments and Pods will be shown in a visual group based on the below label
   ```yaml
   labels:
     app.kubernetes.io/part-of: <GROUP_NAME>
   ```
9. Workload resources like Deployment or Pod will be shown with a visual connector based on the below annotations
   ```yaml
   annotations:
     ...
     app.openshift.io/connects-to: '[{"apiVersion": <RESOURCE_APIVERSION>,"kind": <RESOURCE_KIND>,"name": <RESOURCE_NAME>}]'
   ```

Reply:
@mramendi / @hmanwani-rh have made an attempt to add more details, feel free to modify the wordings / improve. LMK if it makes sense

can read more https://backstage.io/docs/features/kubernetes/configuration/#surfacing-your-kubernetes-components-as-part-of-an-entity

@debsmita1 / @divyanshiGupta FYI

Reply:
Made the suggested changes.